### DISCUSSION TITLE
How to authenticate provider with GitHub deploy key?

### DISCUSSION THREAD

**Discussion by mloskot** [integrations/terraform-provider-github#2032] [Q&A] [answered]
I wonder, why the [current authentication options for the provider](https://registry.terraform.io/providers/integrations/github/5.42.0/docs#authentication) do not allow to authenticate with the GitHub deploy key?

This would be useful if one wants to manage a particular repository only (e.g. add files, etc.) without managing any GitHub resources at level above (i.e. organization, etc.).

**Comment by laughedelic** [comment, marked as answer]
> manage a particular repository only (e.g. add files, etc.) without managing any GitHub resources at level above (i.e. organization, etc.)

You can achieve that with a [fine-grained personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token) (can be limited to a single repo) or by creating a [GitHub app](https://docs.github.com/en/apps/creating-github-apps/about-creating-github-apps/about-creating-github-apps) that is installed in a single repository

**Reply by mloskot** [reply]
Thanks, I will investigate the fine-grained PAT way.
